website/content/docs/discovery/dns.mdx

```diff
@@ -353,32 +393,77 @@ $ echo -n "20010db800010002cafe000000001337" | perl -ne 'printf join(":", unpack
 
 By default, all service lookups use the `default` namespace
 within the partition and datacenter of the Consul agent that received the DNS query.
-To lookup services in another namespace, partition, and/or datacenter,
+To lookup services in another namespace, partition, datacenter, or cluster peer,
 use the [canonical format](#canonical-format).
 
 Consul server agents are in the `default` partition.
 If DNS queries are addressed to Consul server agents,
 service lookups to non-`default` partitions must explicitly specify
 the partition of the target service.
 
-To lookup services imported from a cluster peer,
-refer to [service virtual IP lookups for Consul Enterprise](#service-virtual-ip-lookups-for-consul-enterprise) instead.
+To lookup services directly from a mesh service's application code
+with transparent proxy enabled, refer to 
+[service virtual IP lookups for Consul Enterprise](#service-virtual-ip-lookups-for-consul-enterprise) instead.
 
 #### Canonical format
 
-Use the following query format to specify namespace, partition, and/or datacenter
-for `.service`, `.connect`, `.virtual`, and `.ingress` service lookup types.
-All three fields (`namespace`, `partition`, `datacenter`) are optional.
-```text
-[<tag>.]<service>.service[.<namespace>.ns][.<partition>.ap][.<datacenter>.dc]<domain>
-```
+Use the following query formats to specify namespace, partition,
+WAN federated datacenter, or cluster peer in service lookups.
+In each format, all fields in square brackets are optional.
+
+- Lookup a service, optionally in a cluster peer:
+
+  ```text
+  [<tag>.]<service>.service[.<namespace>.ns][.<peer>.peer][.<partition>.ap].<domain>
+  ```
+
+  This format supports `.service` lookup types in Consul 1.14.2 or later
+  and supports `.virtual` lookup types in Consul 1.14.0 or later.
+
+  The `peer` is a cluster peer of the `partition`.
+  If no `peer` is specified, the service lookup applies to the specified `partition`
+  rather than one of its cluster peers.
+
+  For example, assume that an agent in the `default` partition is acting as the
+  DNS server, the `k8s-app-1` partition is in the same datacenter, and
+  the `k8s-app-1` partition has a cluster peer named `billing`.
+  To perform a lookup of non-mesh service `api` in namespace `foo`
+  of that cluster peer, use the query
+  `api.service.foo.ns.billing.peer.k8s-app-1.partition.consul`.
+
+- Lookup a service, optionally in a WAN federated datacenter:
+
+  ```text
+  [<tag>.]<service>.service[.<namespace>.ns][.<partition>.ap][.<datacenter>.dc].<domain>
+  ```
+
+  This format supports `.service`, `.connect`, `.virtual`, and `.ingress` service lookup types
+  in Consul 1.14.0 or later.
+
+  Cluster peering must be used to connect datacenters containing admin partitions,
+  not WAN federation. Therefore, this format is not applicable to lookup a partition
+  in a different datacenter.
```

Comment:
```suggestion
  You must establish a peer connection between your Consul clusters to connect datacenters containing admin partitions. You cannot query an admin partition in another datacenter over a WAN federated network. 
```

Reply:
The limitation is not on connection / querying. WAN federation and admin partitions are mutually exclusive. If two datacenters are WAN federated, they can't have (non-default) admin partitions.

I'm not sure if that nuance is important here. I'll have to think about whether there's a clearer way than the original text...